spec/lib/fastly_nsq/fake_message_queue_spec.rb

```diff
@@ -37,30 +37,19 @@
 end
 
 RSpec.describe FakeMessageQueue::Producer do
-  after do
-    FakeMessageQueue.reset!
-  end
+  after { FakeMessageQueue.reset! }
 
-  describe '#write' do
-    it 'adds a new message to the queue' do
-      topic = 'death_star'
+  let(:topic)    { 'death_star' }
+  let(:producer) { FakeMessageQueue::Producer.new topic: topic }
 
-      producer = FakeMessageQueue::Producer.new(
-        nsqd: ENV.fetch('NSQD_TCP_ADDRESS'),
-        topic: topic,
-      )
-      producer.write('hello')
+  it 'adds a new message to the queue' do
+    producer.write('hello')
 
-      expect(FakeMessageQueue.queue.size).to eq 1
-    end
+    expect(FakeMessageQueue.queue.size).to eq 1
   end
 
-  describe '#terminate' do
-    it 'has a `terminate` method which is a noop' do
-      producer = instance_double('FakeMessageQueue::Producer')
-
-      allow(producer).to receive(:terminate)
-    end
+  it 'has a `terminate` method which is a noop' do
+    expect(producer.respond_to?(:terminate)).to be_truthy
```

Comment:
This is nice - just enough testing.

I believe since this is a predicate method, you might be able to say `expect(producer).to respond_to(:terminate)`.